### PR TITLE
chore: change form validation

### PR DESCRIPTION
**Pull request by hein-j**
*Description of changes:*
- extend existing validation instead of replacing it if custom validation function
- factor out the validation tasks so that it can be reused on submit later (currently only used on change)
- fix logic for getting if form is valid.
- remove empty error messages on mount

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/codegen-ui-react/lib/forms/form-renderer-helper.ts [hunk 1]
```diff
@@ -451,7 +492,7 @@ export const addFormAttributes = (
   return attributes;
 };
 
-export const buildOnChangeStatement = (
+export const XXbuildOnChafdafdngeStatement = (
```
Comment (Jshhhh) [inline packages/codegen-ui-react/lib/forms/form-renderer-helper.ts:495]: I presume this is slightly unintentional? haha

Comment (hein-j) [reply packages/codegen-ui-react/lib/forms/form-renderer-helper.ts]: d'oh! thanks, deleted the whole block

File: packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react-forms.test.ts.snap [hunk 2]
```diff
@@ -25,24 +25,23 @@ export default function customDataForm(props) {
...
   const [modelFields, setModelFields] = useStateMutationAction({});
   const [formValid, setFormValid] = useStateMutationAction(true);
+  const [errors, setErrors] = useStateMutationAction({});
+  const validations = {
+    name: [],
+    email: [],
+    city: [],
+    category: [],
+  };
+  async function runValidationTasks(fieldName, value) {
```
Comment (SwaySway) [inline packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react-forms.test.ts.snap:37]: nit: can we use an arrow function?

**Review by SwaySway — COMMENTED** [review]
since the errors is starting with an object can we use optional chaining for instances like below?
```
errorMessage={errors.name.errorMessage}
hasError={errors.name.hasError}
```

File: packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react-forms.test.ts.snap (hunk 2 above)
Comment (hein-j) [reply packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react-forms.test.ts.snap]: done

**Comment by hein-j** [comment]
> since the errors is starting with an object can we use optional chaining for instances like below?
> 
> ```
> errorMessage={errors.name.errorMessage}
> hasError={errors.name.hasError}
> ```

good call. done

**Review by SwaySway — APPROVED** [review]

**Review by Jshhhh — APPROVED** [review]
LGTM

**Pull request merged**